self_hosted/main.jou

```diff
@@ -90,6 +90,10 @@ def parse_args(argc: int, argv: byte**) -> CommandLineArgs:
 
     return result
 
+def find_file(files: FileState*, nfiles: int, path: byte*) -> FileState:
+    for i = 0; i < nfiles; i++:
+        if strcmp(files[i].ast.path, path) == 0:
+            return files[i]
```

Comment:
If returning a pointer:
```suggestion
            return &files[i]
    return NULL
```

Reply:
Here `&files[i]` is a pointer to `files[i]` (a `FileState`), not a copy of the whole `FileState`.